flask/app/routes.py

```diff
@@ -506,3 +506,56 @@ def delete_tissue(id: int):
         return "Tissue has dataset(s), cannot delete", 422
     else:
         return "Not Found", 404
+
+@app.route("/api/tissue_samples", methods=["POST"])
+@login_required
+def post_tissue():
+    if not request.json:
+        return "Request body must be JSON", 400
+
+    models.Participant.query.filter_by(participant_id=request.json["participant_id"]).first_or_404()
+
+    enum_error = enum_validate(
+        models.TissueSample,
+        request.json,
+        ["tissue_sample_type","tissue_processing"]
+        )
+
+    if enum_error:
+        return enum_error, 400
+
+    now = datetime.now()
+    created = now
+    updated = now
+
+    try:
+        created_by = updated_by = current_user.user_id
+    except:  # LOGIN DISABLED
+        created_by = updated_by = 1
+
+    tissue_sample = models.TissueSample(
+        **{
+            "participant_id": request.json["participant_id"],
+            "extraction_date": request.json["extraction_date"],
+            "tissue_sample_type": request.json["tissue_sample_type"],
+            "tissue_processing": request.json["tissue_processing"],
+            "notes": request.json["notes"],
```

Comment:
you can use the get method (`request.json.get('notes')`) so `None` is returned if the value for a field isn't supplied in the json. right now if there is no value say for notes then `JSONDecodeError` is raised

Reply:
neat, thanks!